boltrouter/bolt_request.go

```diff
@@ -195,59 +202,84 @@ func (br *BoltRouter) DoBoltRequest(logger *zap.Logger, boltReq *BoltRequest) (*
 	logger.Debug("initial request target", zap.String("target", initialRequestTarget), zap.String("reason", reason))
 
 	if initialRequestTarget == "bolt" {
-		beginTime := time.Now()
-		resp, err := br.boltHttpClient.Do(boltReq.Bolt)
-		duration := time.Since(beginTime)
-		boltRequestAnalytics.BoltRequestDuration = duration
-		if err != nil {
-			if resp != nil {
-				boltRequestAnalytics.BoltRequestResponseStatusCode = resp.StatusCode
-			}
-			return resp, false, boltRequestAnalytics, err
-		} else if !StatusCodeIs2xx(resp.StatusCode) && br.config.Failover {
-			b, _ := io.ReadAll(resp.Body)
-			resp.Body.Close()
-			logger.Error("bolt request failed", zap.Int("statusCode", resp.StatusCode), zap.String("body", string(b)))
-			beginTime := time.Now()
-			resp, err := http.DefaultClient.Do(boltReq.Aws)
-			duration := time.Since(beginTime)
-			boltRequestAnalytics.AwsRequestDuration = duration
-			if err != nil {
-				if resp != nil {
-					boltRequestAnalytics.AwsRequestResponseStatusCode = resp.StatusCode
-				}
-			}
-			return resp, true, boltRequestAnalytics, err
+		resp, isFailoverRequest, err := br.doBoltRequest(logger, boltReq, false, boltRequestAnalytics)
+		// if nothing during br.doBoltRequest panics, err will not be of type ErrPanicDuringBoltRequest so failover was
+		// handled inside the function as needed and we can just return
+		// If the err is of type ErrPanicDuringBoltRequest then we need to failover to AWS manually since .doBoltRequest
+		// halted execution before it could failover
+		if err != nil && err == ErrPanicDuringBoltRequest && br.config.Failover {
```

Comment:
Do we need analytics logs here? Something akin to This request was tried, that resulted in a Panic